recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -20,10 +20,103 @@ processMatch: []
-        exit 132
+
+      if [ $code != "200" ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/credentials ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/config ]; then
+        exit 1
+      fi
+
+      if ! command -v aws &> /dev/null; then
+        exit 1
+      fi
+
+      callerIdentity=$(aws sts get-caller-identity 2>&1)
+
+      if [ "$(echo "$callerIdentity" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+        exit 1
+      fi
+
+      function isUserAdmin {
+        username=$(echo "$1" | sed -ne 's/.*"Arn": ".*\/\(.*\)".*/\1/p')
+
+        userPolicies=$(aws iam list-user-policies --user-name $username 2>&1)
+
+        if [ "$(echo "$userPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+          return 1
+        fi
+
+        attachedUserPolicies=$(aws iam list-attached-user-policies --user-name $username 2>&1)
+
+        if [ "$(echo "$attachedUserPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+          return 1
+        fi
+
+        if [ "$(echo "$attachedUserPolicies" | grep AdministratorAccess | wc -l)" ]; then
+          return 0
+        fi
+
+        groups=$(aws iam list-groups-for-user --user-name $username 2>&1)
+
+        # For each group in groups
+        listOfGroups=$(echo "$groups" | sed -n 's/.*:group\/\(.*\)",.*/\1/p')
+
+        for i in $listOfGroups; do
+          groupPolicies=$(aws iam list-group-policies --group-name $i 2>&1)
+
+          if [ "$(echo "$groupPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+            return 1
+          fi
+
+          attachedGroupPolicies=$(aws iam list-attached-group-policies --group-name $i 2>&1)
+
+          if [ "$(echo "$attachedGroupPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+            return 1
+          fi
+
+          if [ "$(echo "$attachedGroupPolicies" | grep AdministratorAccess | wc -l)" -eq 1 ]; then
+            return 0
+          fi
+
+          groupPolicies=$(aws iam list-group-policies --group-name $i 2>&1)
+
+          if [ "$(echo "$groupPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+            return 1
+          fi
+
+          attachedGroupPolicies=$(aws iam list-attached-group-policies --group-name $i 2>&1)
+
+          if [ "$(echo "$attachedGroupPolicies" | grep AccessDenied | wc -l)" -eq 1 ] ; then
+            return 1
+          fi
+
+          if [ "$(echo "$groupPolicies" | grep AdministratorAccess | wc -l)" -eq 1 ] ; then
+            return 0
+          fi
+
+          if [ "$(echo "$attachedGroupPolicies" | grep AdministratorAccess | wc -l)" -eq 1 ] ; then
+            return 0
+          fi
+        done
+
+        return 1
+      }
+
+      if [ "$(isUserAdmin $callerIdentity)" -eq 1 ]; then
+        exit 1
+      fi
+
+      deployed=$(aws cloudformation describe-stacks --stack-name NewRelic-Metric-Stream 2>&1 | grep "^An error occurred (ValidationError)*" | wc -l)
+
+      if [ "$deployed" -eq "0" ] ; then
+        exit 1
       fi
```

Comment:
@noahmmcgivern it might be good to give a final exit 0 and a comment to indicate success. Here is an example from another recipe:

      # everything looks good!
      exit 0

Reply:
@noahmmcgivern it would be good to have this final exit 0 to be explicit about success to run the recipe